app/models/set_member_subject.rb

```diff
@@ -4,7 +4,6 @@ class SetMemberSubject < ActiveRecord::Base
   
   belongs_to :subject_set, counter_cache: true, touch: true
   belongs_to :subject
-  has_many :classifications
```

Comment:
I'm thinking of removing all the classification counter caches for
classifications and using the Kafka stream to calculate them instead.

The main reason for removing it is needing to handle multi subject
classifications which can't be supported through the normal rails relations
without another join table.
On Jan 29, 2015 8:42 AM, "Campbell Allen" <notifications@github.com> wrote:

> In app/models/set_member_subject.rb
> <https://github.com/zooniverse/Panoptes/pull/458#discussion_r23771279>:
>
> > @@ -4,7 +4,6 @@ class SetMemberSubject < ActiveRecord::Base
> >
> >    belongs_to :subject_set, counter_cache: true, touch: true
> >    belongs_to :subject
> > -  has_many :classifications
>
> I think the main issue with losing this relation is that it had the
> classification counter cache on it. The plan was to use this to cache the
> number of raw classification for a SMS that was then to be used in simple
> retirement (first version). If we lost the counter cache then we'll have to
> query the classification table (possibly very large) and possibly costly as
> time goes on.
>
> We can just keep this relation and get all set_member_subjects for a
> subject_set and then sum the classification counts. Thoughts?
>
> Finally if we do remove it, then the _count field in the schema should go
> to.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/zooniverse/Panoptes/pull/458/files#r23771279>.
>